package/MDAnalysis/core/groups.py

```diff
@@ -3075,11 +3162,8 @@ def atoms(self):
         are further ordered by :class:`Segment` in the :class:`SegmentGroup`.
         Duplicates are *not* removed.
         """
-        # If indices is an empty list np.concatenate will fail (Issue #1999).
-        try:
-            ag = self.universe.atoms[np.concatenate(self.indices)]
-        except ValueError:
-            ag = self.universe.atoms[self.indices]
+        u = self.universe
+        ag = u.atoms[u._topology.tt.segments2atoms_1d(self._ix)]
```

Comment:
This looks ugly, but I know it's because of the design.  I think when I wrote this I liked having the translation table (`tt`) separate so one day we could optimise it....